lib/ddtrace/configuration/settings.rb

```diff
@@ -412,6 +425,7 @@ def initialize(*_)
       end
 
       # [Continuous Integration Visibility](https://docs.datadoghq.com/continuous_integration/) configuration.
+      # @configure_with Datadog::Tracing
       # @public_api
       settings :test_mode do
```

Comment:
No; `test_mode` is here to stay in `Tracing`. It's the counter-part to `ci_mode`.

Users using only tracing can activate `test_mode` to generate some traces in their tests. The purpose of this is to verify tracing is working in their application, particularly if they added manual instrumentation to their code.

Users using CI can activate `ci_mode` which instruments the test runs themselves, for the CI product. That's built upon `test_mode` and extends its functionality a bit for its own purposes.

I think the `trace_flush` and `writer_options` options are still be necessary, and the "remove" may have been added in error.

Reply:
> Users using only tracing can activate test_mode to generate traces in their tests. The purpose of this is to aid verification of tracing in their application, particularly if they added manual instrumentation to their code.

Hmm... It may be worth documenting this use case of `test_mode`: I hadn't realized it was a thing, and I searched our public docs and it doesn't seem to be referenced anywhere.